code/game/objects/items/devices/PDA/cart/misc.dm

```diff
@@ -86,6 +87,63 @@
         charges--
         P.explode(U)
 
+/obj/item/weapon/cartridge/syndifake
+	name = "\improper F.R.A.M.E. Cartridge"
+	icon_state = "cart"
+	origin_tech = Tc_PROGRAMMING + "=2;" + Tc_SYNDICATE + "=2"
+	mech_flags = MECH_SCAN_ILLEGAL
+	starting_apps = list(/datum/pda_app/cart/virus/fake_uplink)
+	var/uses = 0
+
+/obj/item/weapon/cartridge/syndifake/attackby(var/obj/item/W, var/mob/user)
+	if(..())
+		return
+	if(istype(W, /obj/item/stack/telecrystal))
+		var/obj/item/stack/telecrystal/crystals = W
+		uses += crystals.amount
+		to_chat(user, "<span class='notice'>You insert [crystals.amount] telecrystal[crystals.amount > 1 ? "s" : ""] into the cartridge.</span>")
+		crystals.use(crystals.amount)
+
+/obj/item/weapon/cartridge/syndifake/attack_self(var/mob/user)
+	if(..())
+		return
+	if(uses)
+		var/obj/item/stack/telecrystal/crystals = new(user.loc, uses)
+		to_chat(user, "<span class='notice'>You remove [crystals.amount] telecrystal[crystals.amount > 1 ? "s" : ""] from the cartridge.</span>")
+		uses = 0
+		user.put_in_hands(crystals)
+
+/datum/pda_app/cart/virus/fake_uplink
+	name = "Send Uplink"
+	desc = "Frame someone as a tator"
+	icon = "pda_boom"
+	virus_type = "fake uplinks"
+
+/datum/pda_app/cart/virus/fake_uplink/infect(var/obj/item/device/pda/P,var/mob/U)
+	if(P.get_component(/datum/component/uplink))
+		U.show_message("<span class='warning'>An error flashes on your [src]; [pick(syndicate_code_response)]</span>", 1)
+		U << browse(null, "window=pda")
+		var/datum/pda_app/messenger/app = locate(/datum/pda_app/messenger) in P.applications
+		if(app)
+			app.create_message(null, P, null, null, pick(syndicate_code_phrase)) //friendly fire
+		charges--
+	else
+		var/datum/component/uplink/new_uplink = P.add_component(/datum/component/uplink)
+		if(istype(cart_device,/obj/item/weapon/cartridge/syndifake))
+			var/obj/item/weapon/cartridge/syndifake/SF = cart_device
+			new_uplink.telecrystals = SF.uses
+		else
+			new_uplink.telecrystals = 0
+		var/datum/component/uplink/our_uplink = pda_device.get_component(/datum/component/uplink)
+		if(!our_uplink && ismob(pda_device.loc))
+			var/mob/M = pda_device.loc
```

Comment:
Why are you doing any of this instead of using `U`?
Actually, should you be re-using the traitor's unlock code at all? Wouldn't that allow you to verify someone's uplink had been planted if you learned the original code somehow?

Reply:
Made it use U. Plus I don't think there's a way to know the code unless you're the one who has it.

Reply:
Yeah, probably. Maybe changelings can get it? It would be a very fringe situation.

Reply:
Emergent gameplay